### PR TITLE
JAPI-39 Annotation changes in grammar

### PR DESCRIPTION
**Pull request by Michael-Gardner**
This is in addition to the work that Luke Renn did for handling annotations embedded within comments.  I added multi line comment annotations and the appropriate test methods.

@rpastrana please review

### CODE REVIEW

File: wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4 [hunk 1]
```diff
@@ -40,14 +40,14 @@ record_def_inline:
 ;
 
 record_def:
-    REC_SYM (COMMA maxlength)? eclfield_decl SEMI (eclfield_decl SEMI)* END_SYM SEMI
+    REC_SYM (COMMA maxlength)? eclfield_decl SEMI comment? (eclfield_decl SEMI comment?)* END_SYM SEMI
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4]: consider rewriting into something like this:
record_atom:
 (COMMA maxlength)? eclfield_decl SEMI comment?
;
record_def:
REC_SYM  record_atom+ END_SYM SEMI

File: wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4 [hunk 2]
```diff
@@ -40,14 +40,14 @@ record_def_inline:
...
 ;
 defined_record_def :
     TOKEN ASSING_SYM (record_def|record_def_inline)
 ;
 
 exploded_dataset_record_def:
-    REC_SYM (COMMA maxlength)? eclfield_decl SEMI (eclfield_decl SEMI)* END_SYM
+    REC_SYM (COMMA maxlength)? eclfield_decl SEMI comment? (eclfield_decl SEMI comment?)* END_SYM
```
Comment (rpastrana) [inline wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4]: What's the difference between exploded_dataset_record_def and record_def?

**Comment by rpastrana** [comment]
@Michael-Gardner looks good, left a couple of questions/comments on the grammar...

File: wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4 (hunk 2 above)
Comment (Michael-Gardner) [inline wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclRecord.g4]: They look like they're handled the exact same way in the eclreader code.  Perhaps this is something we should look at in the future

**Comment by Michael-Gardner** [comment]
I will be making a new issue to cover the grammar comments you made

**Pull request merged**